### PR TITLE
Annotate Microsoft.CSharp library code for DynamicCode

### PR DESCRIPTION
**Pull request by tlakollo**
Annotates Microsoft.CSharp library code for dangerous Dynamic Code problems flagged by a Roslyn analyzer.
Adding the Microsoft.CSharp label to flag that this PR might create merge conflicts with other files if merged.

### CODE REVIEW

**Comment by cston** [comment]
>         public static bool TryConvert(ConvertBinder binder, DynamicMetaObject instance, out DynamicMetaObject result)

Why is this method not annotated?

---
Refers to: src/libraries/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/ComInterop/ComBinder.cs:199 in 9e2f81b. [](commit_id = 9e2f81b928c403e81404e424e824d0b0438b3207, deletion_comment = False)

**Review by cston — APPROVED** [review]

File: src/libraries/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Binder.cs [hunk 1]
```diff
@@ -17,6 +17,7 @@ namespace Microsoft.CSharp.RuntimeBinder
     public static class Binder
     {
         internal const string TrimmerWarning = "Using dynamic types might cause types or members to be removed by trimmer.";
+        internal const string DynamicCodeWarning = "Using dynamic types will might need runtime code generation.";
```
Comment (vitek-karas) [inline src/libraries/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Binder.cs:20]:
```suggestion
        internal const string DynamicCodeWarning = "Using dynamic types might need runtime code generation.";
```

**Review by vitek-karas — APPROVED** [review]

**Comment by danmoseley** [comment]
what's left here @tlakollo ? also, merge conflict

**Comment by cston** [comment]
@agocke, @tlakollo, is this change intended for .NET 8? Thanks.

**Pull request closed without merging**